File: Game/Content/Datas/Scripts/System/system-item.js
/*
    RPG Paper Maker Copyright (C) 2017 Marie Laporte

    This file is part of RPG Paper Maker.

    RPG Paper Maker is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    RPG Paper Maker is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
*/

// -------------------------------------------------------
//
//  CLASS SystemItem
//
// -------------------------------------------------------

/** @class
*   An item of the game.
*   @property {string} name The name of the item.
*   @property {number} idType The id of the item's type.
*   @property {boolean} consumable Indicate if the item is consumable.
*/
function SystemItem(){

}

SystemItem.prototype = {

    /** Read the JSON associated to the item.
    *   @param {Object} json Json object describing the object.
    */
    readJSON: function(json){
        this.name = json.name;
        this.idType = json.t;
        this.consumable = json.cons;
    }
}

File: Models/System/systemitem.cpp
/*
    RPG Paper Maker Copyright (C) 2017-2019 Marie Laporte

    This file is part of RPG Paper Maker.

    RPG Paper Maker is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    RPG Paper Maker is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
*/

#include "systemitem.h"

// -------------------------------------------------------
//
//  CONSTRUCTOR / DESTRUCTOR / GET / SET
//
// -------------------------------------------------------

SystemItem::SystemItem() :
    SystemCommonSkillItem()
{

}

SystemItem::SystemItem(int i, LangsTranslation *names, int pictureID, int type,
    bool consumable, LangsTranslation *description, TargetKind targetKind,
    PrimitiveValue *targetConditionFormula, AvailableKind availableKind,
    SystemPlaySong *sound,  PrimitiveValue *animationUserID, PrimitiveValue
    *animationTargetID, PrimitiveValue *price, QStandardItemModel *modelEffects) :
    SystemCommonSkillItem (i, names, pictureID, type, consumable, true,
        description, targetKind, targetConditionFormula, new PrimitiveValue(
        PrimitiveValueKind::None), availableKind, sound, animationUserID,
        animationTargetID, price, new QStandardItemModel, modelEffects, new
        QStandardItemModel)
{

}

SystemItem::~SystemItem() {

}

// -------------------------------------------------------
//
//  VIRTUAL FUNCTIONS
//
// -------------------------------------------------------

void SystemItem::read(const QJsonObject &json){
    SystemCommonSkillItem::read(json);
}

// -------------------------------------------------------

void SystemItem::write(QJsonObject &json) const{
    SystemCommonSkillItem::write(json);
}
